src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -388,12 +396,20 @@ void IList.Remove(object value)
 
         void ICollection.CopyTo(Array array, int index)
         {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
-            if (array.Rank != 1)
-                throw new ArgumentException(nameof(array));
+            if ((array != null) && (array.Rank != 1))
+            {
+                throw new ArgumentException(); // Arg_RankMultiDimNotSupported
+            }
 
-            Array.Copy(_items, 0, array, index, _size);
+            try
+            {
+                // Array.Copy will check for null.
+                Array.Copy(_items, 0, array, index, Count);
+            }
+            catch (ArrayTypeMismatchException)
+            {
+                throw new ArgumentException(); // Argument_InvalidArrayType
```

Comment:
@AlexRadch It is easy to add it manually. Just open the XML file ([here](https://github.com/dotnet/corefx/blob/master/src/System.Linq.Expressions/src/Resources/Strings.resx)) with a text editor instead of VS, and follow the format of the other entries at the bottom. e.g.

```xml
  <data name="Argument_InvalidArrayType" xml:space="preserve">
    <value>...</value>
  </data>
```